odl/solvers/util/callback.py

```diff
@@ -483,7 +492,17 @@ def __init__(self, *args, **kwargs):
         self.fig = kwargs.pop('fig', None)
         self.step = kwargs.pop('step', 1)
         self.saveto = kwargs.pop('saveto', None)
+        try:
+            self.saveto_formatter = self.saveto.format
+        except AttributeError:
+            self.saveto_formatter = self.saveto
```

Comment:
My opinion is we stop using newint, I don't see where we need it, do you have a usecase?

We cannot stop using `np.isscalar` here since it is scipy calling it. 

Blacklisting scipy 0.19 is, imo, not an option given that this error only occurs in this really minor function. In that case I'd prefer an x_fail.